packages/@react-aria/menu/docs/useMenu.mdx

```diff
@@ -0,0 +1,442 @@
+`useMenu`, `useMenuItem`, and `useMenuSection` handle exposing this to assistive
+technology using ARIA, along with handling keyboard, mouse, and interactions to support
+selection and focus behavior.
+
+`useMenu` returns props that you should spread onto the menu container element:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType properties={docs.links[docs.exports.useMenu.return.id].properties} />
+</TypeContext.Provider>
+
+`useMenuItem` returns props for an individual item and its children:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType properties={docs.links[docs.exports.useMenuItem.return.id].properties} />
+</TypeContext.Provider>
+
+`useMenuSection` returns props for a section:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType properties={docs.links[docs.exports.useMenuSection.return.id].properties} />
+</TypeContext.Provider>
+
+State is managed by the <TypeLink links={treeDocs.links} type={treeDocs.exports.useTreeState} />
+hook from `@react-stately/tree`. The state object should be passed as an option 
+to each of the above hooks.
+
+If a menu, menu item, or group does not have a visible label, an `aria-label` or `aria-labelledby`
+prop must be passed instead to identify the element to assisitive technology.
+
+## State management
+
+`useMenu` requires knowledge of the items in the menu in order to handle keyboard
+navigation and other interactions. It does this using the <TypeLink links={collectionsDocs.links} type={collectionsDocs.exports.Collection} />
+interface, which is a generic interface to access sequential unique keyed data. You can 
+implement this interface yourself, e.g. by using a prop to pass a list of item objects, 
+but <TypeLink links={treeDocs.links} type={treeDocs.exports.useTreeState} /> from 
+`@react-stately/tree` implements a JSX based interface for building collections instead.
+
+In addition, <TypeLink links={treeDocs.links} type={treeDocs.exports.useTreeState} />
+manages the state necessary for multiple selection and exposes
+a <TypeLink links={selectionDocs.links} type={selectionDocs.exports.SelectionManager} />,
+which makes use of the collection to provide an interface to update the selection state.
+
+## Example
+
+This example uses HTML `<ul>` and `<li>` elements to represent the menu, and applies
+props from <TypeLink links={docs.links} type={docs.exports.useMenu} />
+and <TypeLink links={docs.links} type={docs.exports.useMenuItem} />.
+
+```tsx example
+import {useTreeState} from '@react-stately/tree';
+import {Item} from '@react-stately/collections';
+import {useFocus} from '@react-aria/interactions';
+import {mergeProps} from '@react-aria/utils';
+
+function Menu(props) {
+  // Create state based on the incoming props
+  let state = useTreeState({...props, selectionMode: 'none'});
+
+  // Get props for the menu element
+  let ref = React.useRef();
+  let {menuProps} = useMenu({...props, ref}, state);
+
+  return (
+    <ul
```

Comment:
can we put a width on this so that it doesn't stretch the whole width?

Reply:
maybe a max-width? wouldn't want it to be bigger than the device width if it's hard coded.